src/scripts/main.js

```diff
@@ -1,3 +1,301 @@
 'use strict';
 
-// write code here
+// TABLE VARIABLES
+const table = document.querySelector('table');
+const tHead = table.tHead;
+const tBody = table.tBodies[0];
+
+// SORTING FUNCTIONS AND RELATED OBJECTS
+const valueByIndex = (row, index) => row.cells[index].textContent;
+const salaryToNum = (a) => Number(a.substring(1).split(',').join(''));
+
+const getSortingFunction = (index, order, transformFunc = (x) => x) => {
+  return (a, b) => {
+    const valueA = transformFunc(valueByIndex(a, index));
+    const valueB = transformFunc(valueByIndex(b, index));
+
+    if (typeof valueA === 'string' && typeof valueB === 'string') {
+      return order * valueA.localeCompare(valueB);
+    }
+
+    return order * (valueA - valueB);
+  };
+};
+
+const sortingFunctions = {
+  transformFunctions: {
+    age: parseInt,
+    salary: salaryToNum,
+  },
+
+  get: function (index, colHeader, ascOrder) {
+    const transformFunc = this.transformFunctions[colHeader];
+
+    return ascOrder
+      ? getSortingFunction(index, 1, transformFunc)
+      : getSortingFunction(index, -1, transformFunc);
+  },
+};
```

Comment:
This part of the code is a bit hard to read because of the nested functions and the use of `this`. Consider simplifying this part of the code. You could potentially split it into smaller, more manageable functions.